public/app/features/explore/state/actions.ts

```diff
@@ -481,9 +482,11 @@ export function runQueries(exploreId: ExploreId): ThunkResult<void> {
 
     const newQuerySub = runRequest(datasourceInstance, transaction.request)
       .pipe(
-        map((data: PanelData) => {
-          return preProcessPanelData(data, queryResponse);
-        })
+        map((data: PanelData) => preProcessPanelData(data, queryResponse)),
+        // Simple throttle for live tailing, in case of > 1000 rows per interval we spend about 200ms on processing and
+        // rendering. In case this is optimized this can be tweaked, but also it should be only as fast as user
+        // actually can see what is happening.
+        live ? throttleTime(500) : identity
```

Comment:
Nice!

Reply:
would it make sense to have throttle before preProcessing? So that it can be skipped for throttled events

Reply:
Yes that would make more sense did not realise that after merging